### PR TITLE
[DOC] Add details for setting Kafka broker env vars

### PR DESCRIPTION
**Pull request by tomncooper**
### Type of change

Documentation

### Description

* Added entry to template summary
* Added section on setting env var field in kafkaContainer template

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

File: documentation/book/assembly-customizing-deployments.adoc [hunk 1]
```diff
@@ -22,8 +22,10 @@ include::con-customizing-labels-and-annotations.adoc[leveloffset=+1]
...
 
+include::con-customizing-containers.adoc[leveloffset=+1]
+
 include::con-customizing-image-pull-policy.adoc[leveloffset=+1]
 
 include::con-customizing-pod-disruption-budgets.adoc[leveloffset=+1]
 
-include::proc-customizing-deployments.adoc[leveloffset=+1]
\ No newline at end of file
+include::proc-customizing-deployments.adoc[leveloffset=+1]
```
Comment (ppatierno) [inline documentation/book/assembly-customizing-deployments.adoc]: what's changed here?

File: documentation/book/con-customizing-labels-and-annotations.adoc [hunk 2]
```diff
@@ -27,3 +27,6 @@ template:
 
 The `labels` and `annotations` fields can contain any labels or annotations that do not contain the reserved string `strimzi.io`.
 Labels and annotations containing `strimzi.io` are used internally by {ProductName} and cannot be configured.
+
+It should be noted that the `metadata` data field and therefore the `label` and `annotations` fields are not applicable to container templates such as the `kafkaContainer`.
```
Comment (ppatierno) [inline documentation/book/con-customizing-labels-and-annotations.adoc]: just "`metadata` field" ?

File: documentation/book/assembly-customizing-deployments.adoc (hunk 1 above)
Comment (tomncooper) [reply documentation/book/assembly-customizing-deployments.adoc]: That is a good question?

Comment (scholzj) [reply documentation/book/assembly-customizing-deployments.adoc]: End of line / new line at the end of the file perhaps? It is a whitespace change.

File: documentation/book/con-customizing-containers.adoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-containers-{context}']
+= Customizing Containers
+
+Using the `kafkaContainer` template field you can set environment variables to be applied to the Kafka broker containers.
```
Comment (PaulRMellor) [inline documentation/book/con-customizing-containers.adoc]:
```suggestion
Using the `kafkaContainer` template property, you can set environment variables for Kafka broker containers.
```

File: documentation/book/con-customizing-containers.adoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Environment variables should be supplied under the `env` field as a list of objects with `name` and `value` fields.
```
Comment (PaulRMellor) [inline documentation/book/con-customizing-containers.adoc]:
```suggestion
Environment variables are supplied under the `env` field as a list of objects with `name` and `value` fields. 
```

File: documentation/book/con-customizing-containers.adoc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                env:
+                    - name: TEST_ENV_1
+                      value: test.env.one
+                    - name: TEST_ENV_2
+                      value: test.env.two
+# ...
+----
+
+Environment variables used internally by Strimzi typically (although not always) begin with the `KAFKA_` prefix and should be avoided. 
+If an environment variable is set that is already in use by Strimzi then it will be ignored and a warning will be posted in the log output.
```
Comment (PaulRMellor) [inline documentation/book/con-customizing-containers.adoc]:
```suggestion
If an environment variable is set that is already in use by Strimzi, it is ignored. A warning is posted in the log output.
```

File: documentation/book/assembly-customizing-deployments.adoc [hunk 6]
```diff
@@ -22,8 +22,10 @@ include::con-customizing-labels-and-annotations.adoc[leveloffset=+1]
 
 include::con-customizing-pods.adoc[leveloffset=+1]
 
+include::con-customizing-containers.adoc[leveloffset=+1]
```
Comment (laidan6000) [inline documentation/book/assembly-customizing-deployments.adoc]: Is it worth adding an extra bullet point about Kafka broker environment variables to the "Customizing deployments" assembly? See https://strimzi.io/docs/master/#assembly-customizing-deployments-str

File: documentation/book/con-customizing-containers.adoc [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            kafkaContainer:
+                env:
+                    - name: TEST_ENV_1
+                      value: test.env.one
+                    - name: TEST_ENV_2
+                      value: test.env.two
+# ...
+----
+
+Environment variables used internally by Strimzi typically (although not always) begin with the `KAFKA_` prefix and should be avoided.
```
[the remark this answers is not in the document]
Comment (laidan6000) [reply documentation/book/con-customizing-containers.adoc]: Do not mention the exceptions unless you define them here or provide a link to documentation where they are covered.

File: documentation/book/assembly-customizing-deployments.adoc (hunk 6 above)
Comment (tomncooper) [reply documentation/book/assembly-customizing-deployments.adoc]: I just assumed it would be covered under the `Kafka.spec.kafka`?

**Comment by tomncooper** [comment]
This is hitting the 50min Travis limit again, although the JDK 11 build snuck in under the line.

**Comment by scholzj** [comment]
@tomncooper I did some restarts and Travis is green now. Are the comments from @laidan6000 settled? Should this be merged now?

File: documentation/book/assembly-customizing-deployments.adoc (hunk 6 above)
Comment (laidan6000) [reply documentation/book/assembly-customizing-deployments.adoc]: Ah, yes, you're right.

File: documentation/book/con-customizing-containers.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-containers-{context}']
+= Customizing containers with environment variables
+
+You can set custom environment variables for Kafka broker containers by using the `kafkaContainer` template property in `Kafka.spec.kafka`.
+These environment variables are defined under the `env` field as a list of objects with `name` and `value` fields.
+The following example shows two custom environment variables set for a Kafka broker:
```
Comment (laidan6000) [inline documentation/book/con-customizing-containers.adoc]: Insert a new paragraph here.

**Comment by tomncooper** [comment]
Ok, minor format change and I think this is ready to go!

**Pull request merged**